tsconfig.json

```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es2017",
```

Comment:
cc @tinybike and @stephensprinkle on whether transpiling down to es2017 is acceptable or not.  I can revert this if it isn't acceptable, but the generated code is _way_ easier to read/debug when it is targeting es2017.  I assume that our webpack process will transpile from es2017 to whatever we are actually targeting so this library targeting es2017 features won't hurt, but I want to verify before I merge this.